src/offset.js

```diff
@@ -82,27 +82,28 @@ jQuery.fn.extend({
 				});
 		}
 
-		var docElem, win,
+		var docElem, win, rect,
 			elem = this[ 0 ],
-			box = { top: 0, left: 0 },
 			doc = elem && elem.ownerDocument;
 
 		if ( !doc ) {
```

Comment:
I am not sure I got your message here. If you plan to remove that condition then you can remove it in separate thread. This PR is not about refacroting, but about support ShadowDOM.